src/content/docs/references/docs/optionals_advanced.md

```diff
@@ -0,0 +1,355 @@
+
+    // Unwrap the `Result` from reliable_result1 and reliable_result2
+    if (try reliable_result1 && try reliable_result2 && 5 > 2)
+    {
+        // `reliable_result1` is unwrapped here and can be used as a normal variable
+        io::printfn("reliable_result1: %s", reliable_result1);
+
+        // `reliable_result2` is unwrapped here and can be used as a normal variable
+        io::printfn("reliable_result2: %s", reliable_result2);
+    }
+
+    // ERROR cannot use logical OR `||`
+    // if (try reliable_result1 || try reliable_result2)
+    // {
+    //     io::printn("this can never happen);
+    // }
+}
+```
+
+## Shorthands to work with Optionals
+
+### Getting the `Excuse`
+
+Retrieving the `Excuse` with `if (catch excuse = optional_value) {...}` is not the only way to get the `Excuse` from an Optional, we can use the macro `@catch` instead.
+```c3
+fn void! main(String[] args)
+{
+    int! optional_value = IoError.FILE_NOT_FOUND?;
+    
+    anyfault excuse = @catch(optional_value);
+    if (excuse)
+    {
+        io::printfn("Excuse found: %s", excuse);
+    }
+}
+```
+
+### Check Optional has a `Result` without unwrapping
+
+The `@ok` macro will return `true` if an Optional `Result` is present and `false` if not. Functionally this is equivalent to `!@catch`
+```c3
+fn void! main(String[] args)
+{
+    int! optional_value = 7;
+    
+    bool result_found = @ok(optional_value);
+    assert(result_found == !@catch(optional_value));
+}
+```
+
+### Assert an Optional cannot have a missing `Result` and the program should panic with force unwrap `!!`
+
+If an `Empty` Optional is unexpected or cannot be easily handled we can panic, use sparingly:
+```c3
+int! optional_value = foo_may_error();
+if (catch excuse = optional_value) 
+{
+    unreachable("Unexpected excuse %s", excuse);
+}
+// Now optional_value is a regular variable
+```
+
+The force unwrap operator `!!` will exit the program if a `Excuse` is returned from `foo_may_error()`.
+```c3
+int regular_value = foo_may_error()!!;
+```
+
+## No void! variables
+
+The `void!` type has no possible representation as a variable, and may
+only be a function return type. 
+
+Use `if (catch)` to handle an `Empty` Optional or read the `Excuse`
+```c3
+fn void! test() 
+{
+    return IoError.FILE_NOT_FOUND?;
+}
+
+if (catch excuse = test()) // Retrieve underlying `Excuse`
+{
+    io::printfn("Excuse found: %s", excuse);
+    return excuse?;
+}
+```
+Or use `if (try)` to handle `Result` in Optional
+```c3
+fn void! test() 
+{
+    return;
+}
+
+if (try test()) // Handle `Result` outcome
+{
+    io::printn("successful test()");
+}
+```
+Store the `Excuse` returned from a `void!` function without `if (catch foo = optional_value)`, use the `@catch` macro to convert the result to an `anyfault`:
+```c3
+fn void! test() 
+{
+    return IoError.FILE_NOT_FOUND?;
+}
+
+anyfault excuse = @catch(test());
+```
+
+
+## Default assign when Optional is `Empty` using `??`
```

Comment:
This part should be higher up, as it's important to learn of this operator. Also, the title needs to be improved, "Return a default value when `Empty`" or something.